src/utils/metadata.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+import { getSetting } from 'docc-render/utils/theme-settings';
+import { normalizeAssetUrl } from 'docc-render/utils/assets';
+
+const themeTitle = getSetting(['meta', 'title'], process.env.VUE_APP_TITLE);
+const createMetaTags = ({ title, description, path }) => [
+  {
+    name: 'description',
+    content: description,
+  },
+  {
+    property: 'og:locale',
+    content: 'en_US',
+  },
+  {
+    property: 'og:site_name',
+    content: themeTitle,
+  },
+  {
+    property: 'og:type',
+    content: 'website',
+  },
+  {
+    property: 'og:title',
+    content: title,
+  },
+  {
+    property: 'og:description',
+    content: description,
+  },
+  {
+    property: 'og:url',
+    content: normalizeAssetUrl(path),
+  },
+  {
+    name: 'twitter:card',
+    content: 'summary_large_image',
+  },
+  {
+    name: 'twitter:description',
+    content: description,
+  },
+  {
+    name: 'twitter:title',
+    content: title,
+  },
+  {
+    name: 'twitter:url',
+    content: normalizeAssetUrl(path),
+  },
+];
+const formatTitle = title => [title, themeTitle].filter(Boolean).join(' | ');
+const addMetaTags = (metadata) => {
+  const keys = Object.keys(metadata);
+  // check current metadata
+  const metaQuery = `meta[${keys[0]}="${metadata[keys[0]]}"]`;
+  if (document.querySelector(metaQuery)) {
```

Comment:
Thanks for updating this to make sure we aren't duplicating metadata elements!

This is a very minor note, but I think we should save the `document.querySelector(metaQuery)` result as a variable so that we aren't repeating this selector query multiple times every place it's used.